test_resource_deletion_during_pvc_clone failing in external mode

The test case given below is failing in external mode.
tests/manage/pv_services/pvc_clone/test_resource_deletion_during_pvc_clone.py::TestResourceDeletionDuringPvcClone::test_resource_deletion_during_pvc_clone


```
    def delete_resource(self, resource_id=0):
        pod_ocp = ocp.OCP(
            kind=constants.POD, namespace=config.ENV_DATA["cluster_namespace"]
        )
        if self.cluster_kubeconfig:
            # Setting 'cluster_kubeconfig' attribute to use as the value of the
            # parameter '--kubeconfig' in the 'oc' commands.
            self.resource_obj[
                resource_id
            ].ocp.cluster_kubeconfig = self.cluster_kubeconfig
            pod_ocp.cluster_kubeconfig = self.cluster_kubeconfig
>       self.resource_obj[resource_id].delete(force=True)
E       IndexError: list index out of range

ocs_ci/helpers/disruption_helpers.py:145: IndexError
```

Failed in RUN ID: 1676676006

--- a/tests/manage/pv_services/pvc_clone/test_resource_deletion_during_pvc_clone.py
+++ b/tests/manage/pv_services/pvc_clone/test_resource_deletion_during_pvc_clone.py
@@ -71,9 +71,9 @@ class TestResourceDeletionDuringPvcClone(ManageTest):
             "cephfsplugin_provisioner",
             "cephfsplugin",
             "rbdplugin",
-            "osd",
-            "mgr",
         ]
+        if not config.DEPLOYMENT["external_mode"]:
+            pods_to_delete.extend(["osd", "mgr"])
         executor = ThreadPoolExecutor(max_workers=len(self.pvcs) + len(pods_to_delete))
         disruption_ops = [disruption_helpers.Disruptions() for _ in pods_to_delete]
         file_name = "file_clone"

--- a/tests/manage/pv_services/pvc_snapshot/test_resource_deletion_during_snapshot_restore.py
+++ b/tests/manage/pv_services/pvc_snapshot/test_resource_deletion_during_snapshot_restore.py
@@ -14,6 +14,7 @@ from ocs_ci.framework.testlib import (
 from ocs_ci.ocs.resources.pod import cal_md5sum, verify_data_integrity
 from ocs_ci.helpers import disruption_helpers
 from ocs_ci.helpers.helpers import wait_for_resource_state
+from ocs_ci.framework import config
 
 log = logging.getLogger(__name__)
 
@@ -56,9 +57,9 @@ class TestResourceDeletionDuringSnapshotRestore(ManageTest):
             "cephfsplugin_provisioner",
             "cephfsplugin",
             "rbdplugin",
-            "osd",
-            "mgr",
         ]
+        if not config.DEPLOYMENT["external_mode"]:
+            pods_to_delete.extend(["osd", "mgr"])
         executor = ThreadPoolExecutor(max_workers=len(self.pvcs) + len(pods_to_delete))
         disruption_ops = [disruption_helpers.Disruptions() for _ in pods_to_delete]
         file_name = "file_snap"

